svc/GridExportService.js

```diff
@@ -181,21 +189,26 @@ export class GridExportService extends HoistService {
             cellHasExcelFormat = isFunction(excelFormat);
 
         if (cellHasExcelFormat) {
-            excelFormat = excelFormat(value, {record, column, gridModel});
+            excelFormat = excelFormat(value, {record, column, gridModel}) ?? ExcelFormat.DEFAULT;
         }
 
-        // 2) Dates: Provide date data expected by server endpoint.
-        // Also, approximate formats for CSV and clipboard.
-        if (excelFormat === ExcelFormat.DATE_FMT) value = fmtDate(value);
-        if (excelFormat === ExcelFormat.DATETIME_FMT) value = fmtDate(value, 'YYYY-MM-DD HH:mm:ss');
+        // 2) Dates: Provide the date data string expected by the server endpoint
+        // Also functions as a consistent human-friendly date format for CSV and clipboard
+        if (type === DATE) value = fmtDate(value, 'YYYY-MM-DD HH:mm:ss');
 
         value = value.toString();
 
-        return forExcel && cellHasExcelFormat ?
-            {value, format: excelFormat} :
+        const cellHasVariedType = type !== fieldType;
+
+        // Send format and/or type with the cell in an object only if it varies within the column
+        return (forExcel && (cellHasExcelFormat || cellHasVariedType)) ?
+            omitBy({
+                value,
+                format: cellHasExcelFormat ? excelFormat : null,
+                type: cellHasVariedType ? type : null
```

Comment:
I don't think we will ever have a cell with a different type to the other cells in the column. I think we can simplify considerably by not considering 'type' at the cell level at all.

This code anticipates a cell having a varied type if the column's `exportValue` points to another field - but those cases are essentially pointing the **entire column** to another field. The new field type would apply to **every** cell in the column, and we can send the other field's type with the column's metadata rather than here.